sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_sync_token.py

```diff
@@ -0,0 +1,106 @@
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+"""
+This module is the requests implementation of Pipeline ABC
+"""
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+class SyncToken(object):
+    """The sync token structure
+        """
+    def __init__(self, token_id, value, sequence_number):
+        self.token_id = token_id
+        self.value = value
+        self.sequence_number = sequence_number
+
+    def __str__(self):
+        return "{}={}".format(self.token_id, self.value)
+
+    @classmethod
+    def from_sync_token_string(cls, sync_token):
+        try:
+            position = sync_token.index(';sn=')
+            sequence_number = int(sync_token[position+4:])
+            id_value = sync_token[:position]
+            position = id_value.index('=')
+            token_id = id_value[:position]
+            value = id_value[position+1:]
+            return SyncToken(token_id, value, sequence_number)
+        except ValueError:
+            return None
+
+class SyncTokenPolicy(SansIOHTTPPolicy):
+    """A simple policy that enable the given callback
+    with the response.
+
+    :keyword callback raw_response_hook: Callback function. Will be invoked on response.
+    """
+    def __init__(self, **kwargs): # pylint: disable=unused-argument,super-init-not-called
+        self._sync_token_header = "Sync-Token"
+        self._sync_tokens = dict()
+
+    def on_request(self, request): # type: ignore # pylint: disable=arguments-differ
+        # type: (PipelineRequest) -> None
+        """This is executed before sending the request to the next policy.
+
+        :param request: The PipelineRequest object.
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        sync_token_header = ''
+        for sync_token in self._sync_tokens.values():
+            if sync_token_header != '':
+                sync_token_header = sync_token_header + ',' + sync_token.__str__()
```

Comment:
Can use += here tho calling the LHS here `sync_token_header` also seems a bit confusing to me, since (header + something else) seems it would no longer be something called "header". 

Also why `x.__str__()`  and not `str(x)`?